server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -58,258 +47,26 @@ public class JoinPlanBuilder {
 
     static LogicalPlan buildJoinTree(List<AnalyzedRelation> from,
                                      Symbol whereClause,
-                                     List<JoinPair> joinPairs,
                                      SubQueries subQueries,
-                                     Function<AnalyzedRelation, LogicalPlan> plan) {
+                                     Function<AnalyzedRelation, LogicalPlan> toLogicalPlan) {
+
         if (from.size() == 1) {
-            LogicalPlan source = subQueries.applyCorrelatedJoin(plan.apply(from.get(0)));
+            LogicalPlan source = subQueries.applyCorrelatedJoin(toLogicalPlan.apply(from.getFirst()));
             return Filter.create(source, whereClause);
         }
-        Map<Set<RelationName>, Symbol> queryParts = QuerySplitter.split(whereClause);
-        List<JoinPair> allJoinPairs = convertImplicitJoinConditionsToJoinPairs(joinPairs, queryParts);
-        LinkedHashMap<Set<RelationName>, JoinPair> joinPairsByRelations = buildRelationsToJoinPairsMap(allJoinPairs);
-
-        Map<RelationName, AnalyzedRelation> relationsFromClause = new HashMap<>();
-        List<RelationName> relationNamesFromClause = new ArrayList<>();
-
+        // We have more than 1 relation, this means we have implicit joins in the query
```

Comment:
This is the only part left to create the implicit joins. Everything else is already built.